framework/src/base/FEProblem.C

```diff
@@ -98,6 +98,93 @@ InputParameters validParams<FEProblem>()
   return params;
 }
 
+//
+InputParameters
+FEProblem::executionerValidParams()
```

Comment:
I understand what you're up to here... but having returned to this after a few months it still feels like `FEProblem` is not the right place for this stuff.

It really feels like we need a `SolverParams` object... and this function would go there.